### PR TITLE
[#1891] Update Cypress to v10

### PR DESCRIPTION
**Pull request by vvidday**
Part of #1891

### Proposed commit message

```
Currently, we are using Cypress 9, which is 3 major versions behind 
the latest release. Older versions are less likely to be supported and
maintained.

Let’s update Cypress to version 10.
```

### Other information

Test files are renamed according to the [new recommended spec pattern](https://docs.cypress.io/guides/references/migration-guide#testFiles--specPattern) in Cypress 10.

The old configuration file has been migrated according to the [new changes](https://docs.cypress.io/guides/references/migration-guide#Configuration-File-Changes).

### CODE REVIEW

**Review by ckcherry23 — COMMENTED** [review]
The code changes look great! 
However, some parts of the developer guide need to be updated (for example: https://reposense.org/dg/workflow.html#testing-front-end) and also the new changes from the `master` branch can be merged.

**Comment by vvidday** [comment]
> The code changes look great! However, some parts of the developer guide need to be updated (for example: https://reposense.org/dg/workflow.html#testing-front-end) and also the new changes from the `master` branch can be merged.

Great point! Just updated the DG with the new screenshots/phrasing (the new UI is the same for v10-v12, minus the version number). Regarding the "Run all specs" portion in the DG, I removed it as the functionality was removed in Cypress 10: https://docs.cypress.io/guides/references/migration-guide#Cypress-Changes

**Review by ckcherry23 — APPROVED** [review]
LGTM!

File: docs/dg/workflow.md [hunk 1]
```diff
@@ -99,9 +99,11 @@ See [vue-devtools project home](https://github.com/vuejs/vue-devtools) page for
 ### Writing tests
 1. Create a new test file in `frontend/cypress/tests`.
 1. At project root start *Cypress Test Runner* by running `gradlew cypress`.
-1. On the top right hand corner, set `Chrome` as the default browser.
-1. Under **Integration Tests**, click on the newly created test file to run it.
-![Cypress Test Runner](../images/cypress-test-runner.jpg "Cypress Test Runner")
+1. Select **E2E Testing** and choose **Chrome** as the preferred browser.
+ ![Cypress Test E2E](../images/cypress-test-e2e.png)
+ ![Cypress Test Chrome](../images/cypress-test-chrome.png)
```
Comment (HCY123902) [inline docs/dg/workflow.md:104]: Can you align the indentations of the image links here

**Review by HCY123902 — APPROVED** [review]

File: docs/dg/workflow.md (hunk 1 above)
Comment (HCY123902) [inline docs/dg/workflow.md:104]: Can you align the indentations of the image links here

**Review by HCY123902 — CHANGES_REQUESTED** [review]

File: docs/dg/workflow.md (hunk 1 above)
Comment (vvidday) [reply docs/dg/workflow.md:104]: Sorry, are you referring to the 3rd image? From my side the first 2 images look aligned:

<details>
<summary>Screenshot</summary>

alignment
</details>

Comment (HCY123902) [reply docs/dg/workflow.md:104]: Oh I mean the indentation of the text in the file itself

Comment (vvidday) [reply docs/dg/workflow.md:104]: Oh got it! Sorry for the misunderstanding - will fix this

Comment (vvidday) [reply docs/dg/workflow.md:104]: > Oh I mean the indentation of the text in the file itself. The number of whitespaces is different from that of other image links here

Just noticed that the images in lines 84/86 have 3 whitespaces of indentation, but the previous links in this section had none - is there a specific one I should follow? I couldn't find anything in the style guide relating to this

Comment (HCY123902) [reply docs/dg/workflow.md:104]: I think you can just remove the preceding whitespaces, in the same way as how the multiline content in a list is indented

**Review by HCY123902 — APPROVED** [review]

**Review by yhtMinceraft1010X — APPROVED** [review]
LGTM

**Pull request merged**